packages/utilities/src/lib/sleep.ts

```diff
@@ -0,0 +1,21 @@
+/**
+ * Sleeps for the specified number of milliseconds.
+ * @param ms The number of milliseconds to sleep.
+ * @param value The value to resolve the promise with.
+ * @see {@link sleepSync} for a synchronous version.
+ */
+export function sleep<T = undefined>(ms: number, value?: T): Promise<T> {
```

Comment:
Shouldn't we at least try to support `AbortSignal` like `timers/promises` does?

https://nodejs.org/dist/latest-v18.x/docs/api/timers.html#timerspromisessettimeoutdelay-value-options

Also the option to unref the timeout, of course, although that would be specific to Node.js, since timers in browsers are just an integer rather than a Timer instance. We can just make it do nothing if the timer is a number.